lib/runner/variant-reassignment.js

```diff
@@ -332,10 +333,44 @@ export default class VariantReassignment {
 
   /**
    * Verify that there are no other products in the platform that has slugs of productDraft
-   * except ctpProductToUpdate.
-   * This method should cover the test variant-reassignment-example-7.spec.js
+   * except ctpProductToUpdate. It's enough to check matchingProducts because it's not probable
+   * that there will be a product which has no matching variant, but a conflicting slug.
+   *
+   * @see variant-reassignment-example-7.spec.js
+   */
+  async _ensureSlugUniqueness (productDraft, matchingProducts) {
+    const productDraftSlug = productDraft.slug
+    const productsToAnonymize = []
+    matchingProducts.forEach((product) => {
+      const isSlugConflicting = this._isSlugConflicting(product, productDraftSlug)
+      if (isSlugConflicting)
+        productsToAnonymize.push(product)
+    })
```

Comment:
Simplified version:
```js
const productsToAnonymize = matchingProducts.map(product =>
  this._isSlugConflicting(product, productDraftSlug)
)
```

Reply:
Do you mean this?
```js
const productsToAnonymize = matchingProducts.filter(product =>
      this._isSlugConflicting(product, productDraftSlug)
    )
```

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/37/commits/418eb9261e545b390510ddcaf12a97939aa61d93